File: src/api/scenes.ts
import apiClient from './client';
import type { Scene, CreateSceneRequest, UpdateSceneRequest, SceneOrder } from '@/types/scene';

export async function createScene(projectId: number, data: CreateSceneRequest): Promise<Scene> {
  return apiClient.post(`/projects/${projectId}/scenes`, data);
}

export async function updateScene(projectId: number, sceneId: number, data: UpdateSceneRequest): Promise<Scene> {
  return apiClient.put(`/projects/${projectId}/scenes/${sceneId}`, data);
}

export async function deleteScene(projectId: number, sceneId: number): Promise<void> {
  return apiClient.delete(`/projects/${projectId}/scenes/${sceneId}`);
}

export async function reorderScenes(projectId: number, sceneOrders: SceneOrder[]): Promise<void> {
  return apiClient.put(`/projects/${projectId}/scenes/reorder`, { scene_orders: sceneOrders });
}

File: backend/app/routers/scenes.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.dependencies import get_current_user
from app.models.user import User
from app.schemas.common import ApiResponse
from app.schemas.scene import (
    CreateSceneRequest,
    ReorderScenesRequest,
    SceneOut,
    UpdateSceneRequest,
)
from app.services import scene_service

router = APIRouter(prefix="/projects/{project_id}/scenes", tags=["Scenes"])


@router.post("", response_model=ApiResponse[SceneOut])
async def create_scene(
    project_id: int,
    body: CreateSceneRequest,
    db: AsyncSession = Depends(get_db),
    _user: User = Depends(get_current_user),
):
    scene = await scene_service.create_scene(db, project_id, body)
    return ApiResponse(data=SceneOut.model_validate(scene))


@router.put("/{scene_id}", response_model=ApiResponse[SceneOut])
async def update_scene(
    project_id: int,
    scene_id: int,
    body: UpdateSceneRequest,
    db: AsyncSession = Depends(get_db),
    _user: User = Depends(get_current_user),
):
    scene = await scene_service.update_scene(db, project_id, scene_id, body)
    return ApiResponse(data=SceneOut.model_validate(scene))


@router.delete("/{scene_id}", response_model=ApiResponse[None])
async def delete_scene(
    project_id: int,
    scene_id: int,
    db: AsyncSession = Depends(get_db),
    _user: User = Depends(get_current_user),
):
    await scene_service.delete_scene(db, project_id, scene_id)
    return ApiResponse(message="Scene deleted")


@router.put("/reorder", response_model=ApiResponse[None])
async def reorder_scenes(
    project_id: int,
    body: ReorderScenesRequest,
    db: AsyncSession = Depends(get_db),
    _user: User = Depends(get_current_user),
):
    await scene_service.reorder_scenes(db, project_id, body.scene_orders)
    return ApiResponse(message="Scenes reordered")
